docs/changes/modeling/15215.bugfix.rst

```diff
@@ -0,0 +1,2 @@
+Apply filtering for non-finite values in either/both the data values
+and the weights for the ``LevMarLSQFitter``.
```

Comment:
Is it? My main motivation for calling it a bugfix is having a released fix ASAP for work in https://github.com/spacetelescope/romancal/pull/794.

Reply:
@pllim do you know of any plans to make an astropy release between now and the astropy 6 release?

Reply:
`romancal` should not dictate `astropy` release policy. 

1. Is this a real bug? 
2. Why is it so urgent for romancal that it cannot wait a few more months for v6 ?

Reply:
And we just did a bugfix like last week. We cannot do it every week because of romancal or Jdaviz . That is not a realistic expectation.

Reply:
> any plans

No concrete plans but it is possible. But I do not like people try to bend the rules here, if that is indeed what is happening.

Reply:
Ultimately, I think it is `modeling` maintainer's call. I would have to trust your judgement as a maintainer, but I hope you would respect the astropy policy and not have romancal cloud your judgement.

Reply:
@pllim I don't see this as bending any rules because I disagree with @WilliamJamieson's position that this is a new feature. I'm discussing with him directly.

Reply:
If the current code crashes I would consider this a bug fix.

Reply:
In the change log, please add a few more words on what buggy behavior this PR is fixing. It currently reads like "API change" entry.

And just how backward compatible is this? You might want this, but are you sure everyone else does?

Reply:
Log edited [here](https://github.com/astropy/astropy/pull/15215/commits/308e81a9cb4284dfebf322a87a0993d69e60f970). 

After this PR, an error will no longer be raised for users who call `LevMarLSQFitter` with non-finite weights and `filter_non_finite=True`, and a fit will be produced. Users who provide finite weights or don't provide `filter_non_finite=True` will see no difference.